### ISSUE TITLE
Shield ssl settings.

### ISSUE THREAD

**Issue by AbrohamLincoln** [elastic/puppet-elasticsearch#629]
I'm attempting to add shield configurations to elasticsearch.yml that allow for client cert auth and I've been running into an issue.

From the docs I've determined I need two settings to meet my goals:
```
shield.http.ssl: true
shield.http.ssl.client.auth: optional
```

What gets written to elasticsearch.yml using this module looks like this:
```
shield:
  http:
    ssl:
      client:
        auth: optional
```

From what I can tell both are valid yaml, but the second results in ES starting without ssl support.  I've found no way to work around this issue using this module.

**Comment by tylerjl** [comment]
Hmmm, I think this makes sense given that the erb attempts to break apart the configuration into a nested hash (i.e. `foo.bar: baz` becomes `{ foo: { bar: 'baz' } }`. The later config option pushes out the `ssl: true` key in that case since the hash key for `ssl` gets appended later.

There's two options here, since we can no longer assume that config file hash options are limited to leafs in the yaml:

* Attempt to detect in the erb code where values aren't a terminus and qualify them with dots in the yaml to be able to do this.
* Change the erb entirely such that it writes out *all* configuration options in the fully-qualified dot format instead of nested yaml key/value hashes.

Not sure which option is preferable; 1) is some pretty specific handling that could be brittle and 2) means anything downstream that consumes the config as yaml breaks.

**Comment by tylerjl** [comment]
@AbrohamLincoln I've got a tentative fix in PR #630, any feedback?

**Comment by AbrohamLincoln** [comment]
@tylerjl Works for me.  Thanks!

**Comment by tylerjl** [comment]
Fixed in #630 (merged)
